functions/Get-DbaDatabase.ps1

```diff
@@ -74,181 +76,152 @@ Returns only the system databases on the local default SQL Server instance
-
-			if ($NoFullBackup -or $NoFullBackupSince -or $NoLogBackup -or $NoLogBackupSince)
-			{
-				$defaults += ('Notes')
-			}
-			foreach ($db in $inputobject)
-			{
-			
-				$Notes = $null
-				if ($NoFullBackup -or $NoFullBackupSince)
-				{		
-					if	(@($db.EnumBackupSets()).count -eq @($db.EnumBackupSets() | Where-Object{$_.IsCopyOnly}).count -and (@($db.EnumBackupSets()).count -gt 0) )
-					{
-						$Notes = "Only CopyOnly backups"
-					}	
-				}	
-				Add-Member -InputObject $db -MemberType NoteProperty BackupStatus -value $Notes
-	
-				Add-Member -InputObject $db -MemberType NoteProperty ComputerName -value $server.NetName
-				Add-Member -InputObject $db -MemberType NoteProperty InstanceName -value $server.ServiceName
-				Add-Member -InputObject $db -MemberType NoteProperty SqlInstance -value $server.DomainInstanceName
-				Select-DefaultView -InputObject $db -Property $defaults
-				
-			}
-		}
-	}
+    [CmdletBinding(DefaultParameterSetName = "Default")]
+    Param (
+        [parameter(Position = 0, Mandatory = $true, ValueFromPipeline = $True)]
+        [Alias("ServerInstance", "SqlServer")]
+        [object[]]$SqlInstance,
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [Alias("SystemDbOnly")]
+        [parameter(ParameterSetName = "NoUserDb")]
+        [switch]$NoUserDb,
+        [Alias("UserDbOnly")]
+        [parameter(ParameterSetName = "NoSystemDb")]
+        [switch]$NoSystemDb,
+        [parameter(ParameterSetName = "DbBackuOwner")]
+        [string[]]$Owner,
+        [parameter(ParameterSetName = "Encrypted")]
+        [switch]$Encrypted,
+        [parameter(ParameterSetName = "Status")]
+        [ValidateSet('EmergencyMode', 'Normal', 'Offline', 'Recovering', 'Restoring', 'Standby', 'Suspect')]
+        [string]$Status,
+        [parameter(ParameterSetName = "Access")]
+        [ValidateSet('ReadOnly', 'ReadWrite')]
+        [string]$Access,
+        [parameter(ParameterSetName = "RecoveryModel")]
+        [ValidateSet('Full', 'Simple', 'BulkLogged')]
+        [string]$RecoveryModel,
+        [switch]$NoFullBackup,
+        [datetime]$NoFullBackupSince,
+        [switch]$NoLogBackup,
+        [datetime]$NoLogBackupSince,
+        [switch]$Silent
+    )
+
+    DynamicParam { if ($SqlInstance) { return Get-ParamSqlDatabases -SqlServer $SqlInstance[0] -SqlCredential $SqlCredential } }
+
+    BEGIN {
+        $databases = $psboundparameters.Databases
+        $exclude = $psboundparameters.Exclude
+
+        if ($NoUserDb -and $NoSystemDb) {
+            Stop-Function -Message "You cannot specify both NoUserDb and NoSystemDb" -Continue -Silent $Silent
+        }
+    }
+
+    PROCESS {
+        foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-SqlServer -SqlServer $instance -SqlCredential $sqlcredential
+            }
+            catch {
+                Stop-Function -Message "Failed to connect to: $instance" -Continue -Silent $Silent
```

Comment:
To improve troubleshooting, add:
`-InnerErrorRecord $_ -Target $instance`
This will include the original exception and the instance for which it failed to the error log